lib/streamlit/hashing.py

```diff
@@ -296,8 +315,11 @@ def _to_bytes(self, obj, context):
                     return self.to_bytes("%s.%s" % (obj.__module__, obj.__name__))
 
                 h = hashlib.new(self.name)
-                # TODO: This may be too restrictive for libraries in development.
-                if os.path.abspath(obj.__code__.co_filename).startswith(os.getcwd()):
+                filepath = os.path.abspath(obj.__code__.co_filename)
+
+                if util.file_is_in_folder(
+                    filepath, _get_main_script_directory()
+                ) and not self._folder_black_list.is_blacklisted(filepath):
```

Comment:
I am separately checking 
- whether the file is in the current directory
- whether the file is not in the blacklist